### ISSUE TITLE
In POSIX sh, cd flags besides -L and -P are undefined

### ISSUE THREAD

**Issue by contivero** [koalaman/shellcheck#1457]
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
set -e
cd -P -e dir
```

#### Here's what shellcheck currently says:
Nothing

#### Here's what I wanted or expected to see:
"In POSIX sh, cd -e is undefined"

The only allowed flags under POSIX sh for the`cd` builtin are `-L` and `-P` ([see spec](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/cd.html)). Dash also follows this. Bash, however, allows: `[-L|[-P [-e]] [-@]`.
